File: pubsub_go/received/setupRedis.go
package main

import (
	"context"

	"github.com/redis/go-redis/v9"
)

func setupRedis() (context.Context, *redis.Client, string) {
	ctx := context.Background()
	channelName := "local_channel"

	rdc := redis.NewClient(&redis.Options{
		Addr: "127.0.0.1:6379",
	})

	return ctx, rdc, channelName
}

File: pubsub_rust/received/src/setup_redis.rs
use redis::Client;

pub fn setup_redis() -> redis::RedisResult<(Client, String)> {
    let channel_name = "local_channel".to_string();
    let client = Client::open("redis://127.0.0.1:6379")?;
    
    Ok((client, channel_name))
}